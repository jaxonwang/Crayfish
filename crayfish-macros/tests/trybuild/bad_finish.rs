use crayfish_macros::*;


pub fn main() {

    let _ = async {
        finish!{
            let a = Some(1);
            let b = a.unwrap()?;
            println("{}", b);
        }
    };
}

