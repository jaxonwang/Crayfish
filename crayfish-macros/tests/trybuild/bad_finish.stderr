error: try expression is not allowed in finish block. TODO: this check might be false positive.
 --> $DIR/bad_finish.rs:9:31
  |
9 |             let b = a.unwrap()?;
  |                               ^
